### PR TITLE
task/WA-157 - STKO Express V3

### PR DESCRIPTION
**Pull request by fnets**
## Overview: ##
Converting STKO Express to run in Tapis V3

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [WA-157](https://tacc-main.atlassian.net/browse/WA-157)

## Summary of Changes: ##
Created a new App Definition and tapisjob_app file to work with STKO Express V3 and the ds-stko-dev system

## Testing Steps: ##
1. Submit a job at : https://dev.cep.tacc.utexas.edu/workbench/applications/stko-express-1.0.0
2. Open Session from jobs history.
3. STKO opens and window is adjustable to any size within the DCV session desktop.
4. No errors (red banners) shown in UI.

### CODE REVIEW

File: systems/ds-stko-dev.json [hunk 1]
```diff
@@ -31,4 +31,4 @@
     "jobCapabilities": [],
     "tags": [],
     "notes": {}
-}
+}
```
Comment (chandra-tacc) [inline systems/ds-stko-dev.json:34]: This file has conflicts according to PR.

**Review by chandra-tacc — CHANGES_REQUESTED** [review]
Few things I noticed when testing:
1. The app says it is not licensed.
Screenshot 2024-05-04 at 8 37 59 AM

2. The window is not expanding beyond a certain dimension. This is the max I could get.
Screenshot 2024-05-04 at 8 38 39 AM

**Comment by fnets** [comment]
> Few things I noticed when testing:
> 
> 1. The app says it is not licensed.
>    Screenshot 2024-05-04 at 8 37 59 AM
> 2. The window is not expanding beyond a certain dimension. This is the max I could get.
>    Screenshot 2024-05-04 at 8 38 39 AM

Licensing issue is known, working with the developers to get a way to access their licensing servers. Issue with window size is known, is the scope of another ticket.

File: applications/stko-express/app.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    "inputMode": "FIXED",
+                    "arg": "--tapis-profile tacc-apptainer",
+                    "notes": {
+                        "isHidden": true
+                    }
+                }
+            ],
+            "envVariables": [
+                {
+                    "key": "_XTERM_CMD",
```
[the remark this answers is not in the document]
Comment (fnets) [reply applications/stko-express/app.json:46]: Good catches on all these. I'll update in a second.

File: systems/ds-stko-dev.json (hunk 1 above)
Comment (fnets) [reply systems/ds-stko-dev.json:34]: Very strange, I'll resolve it here in a second.

**Review by chandra-tacc — APPROVED** [review]
Thanks for addressing comments. LGTM.

**Pull request merged**